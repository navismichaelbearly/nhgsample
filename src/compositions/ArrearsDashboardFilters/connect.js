import { connect } from 'react-redux';

import {
  getStatuses as getArrearStatuses,
  updateFiltersAndFetchResults,
} from '../../ducks/arrears';
import {
  getTasksSummary,
  getStatuses as getTaskStatuses,
  getMyTasksStats,
  getUserTasksSummary,
  getSummaryTasks,
} from '../../ducks/tasks';
import getHelpers from '../../util/stateHelpers';
import { sortPatchList, flattenPatches, flattenUsers } from '../../util/patch';

export const mapStateToProps = state => {
  const { get, getString } = getHelpers(state);

  const profile = get(['user', 'profile']);

  return {
    taskStatuses: get(['tasks', 'statuses']),
    arrearStatuses: get(['arrears', 'statuses']),
    arrearsDashboardLabels: getString(['arrearsDashboard']),
    filterLabels: get(['dictionary', 'arrearsDashboard', 'filters']),
    heading: getString(['arrearsDashboard', 'heading']),
    loadingArrears: get(['arrears', 'loadingArrears']),
    loadingTasks: get(['tasks', 'loading']),
    patches: sortPatchList(get(['patch', 'patchList']), profile),
    myTasks: get(['tasks', 'mySummaryItems']),
    myTasksStats: get(['tasks', 'myTasksStats']),
    mySummaryItemsCount: get(['tasks', 'mySummaryItems', 'count'], 0),
    selectedUserTasksCount: get(['tasks', 'userItemsCount'], 0),
    profile,
    youText: get(['patchList', 'you']),
  };
};

export const mergeProps = (
  { breadcrumb, heading, patches, profile, ...stateProps },
  { dispatch },
  ownProps
) => {
  const getSummary = (opts, isMine /* used when querying for 'My tasks' */) =>
    dispatch(
      getTasksSummary(
        {
          statuses: 'open',
          EntityType: 'Arrears',
          ...opts,
        },
        isMine
      )
    );
  const flattedUsers = flattenUsers(patches)
    ? flattenUsers(patches)
        .split(',')
        .filter(e => e !== profile.id)
    : [];
  const params = {
    owners: flattedUsers.toString(),
    statuses: 'open',
    EntityType: 'Arrears',
    pageSize: 6,
  };
  const flattenPatchesValues = flattenPatches(patches);
  return {
    ...stateProps,
    patches,
    profile,
    getArrearsSummary: (opts, item) => {
      dispatch(updateFiltersAndFetchResults({ patches: flattenPatchesValues, ...opts }, item));
    },
    getArrearStatuses: opts => {
      dispatch(getArrearStatuses({ patches: flattenPatchesValues, ...opts }));
    },
    getTaskStatuses: opts =>
      dispatch(getTaskStatuses({ patches: flattenPatchesValues, EntityType: 'Arrears', ...opts })),
    getSummaryTasks: opts => {
      dispatch(getSummaryTasks({ patches: flattenPatchesValues, EntityType: 'Arrears', ...opts }));
    },
    getMyTasksStats: opts => {
      dispatch(getMyTasksStats({ EntityType: 'Arrears', ...opts }));
    },
    getMyTasksSummary: () => getSummary({ userId: profile.id }),
    getSelectedUserTasks: () => {
      dispatch(getUserTasksSummary({ ...params }));
    },
    getTasksSummary: (opts = {}, isMine) => {
      getSummary({ patches: flattenPatchesValues, ...opts }, isMine);
    },
    ...ownProps,
  };
};

export default connect(mapStateToProps, null, mergeProps);
