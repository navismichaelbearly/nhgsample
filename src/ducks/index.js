import { combineReducers, createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { enableReduxDevtool } from 'nhh-styles';

import activities from './activities';
import arrears from './arrears';
import sendCorrespondence from './sendCorrespondence';
import customer from './customer';
import dictionary from './dictionary';
import features from './features';
import forms from './forms';
import housingofficersearch from './housingofficersearch';
import interactions from './interactions';
import legalReferral from './legalReferral';
import linkedCases from './linkedCases';
import media from './media';
import modal from './modal';
import navigation from './navigation';
import notes from './notes';
import notifications from './notifications';
import patch from './patch';
import pause from './pause';
import paymentPlan from './paymentPlan';
import referral from './referral';
import ribbon from './ribbon';
import tasks from './tasks';
import tenancy from './tenancy';
import user from './user';
import welfareReform from './welfareReform';

export const reducers = combineReducers({
  activities,
  arrears,
  sendCorrespondence,
  customer,
  dictionary,
  features,
  forms,
  housingofficersearch,
  interactions,
  legalReferral,
  linkedCases,
  media,
  modal,
  navigation,
  notes,
  notifications,
  patch,
  pause,
  paymentPlan,
  referral,
  ribbon,
  tasks,
  tenancy,
  user,
  welfareReform,
});

const storeEnhancer = compose(applyMiddleware(thunk), enableReduxDevtool);

export default createStore(reducers, storeEnhancer);
