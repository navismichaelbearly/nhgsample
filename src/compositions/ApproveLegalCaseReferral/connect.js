import { connect } from 'react-redux';
import path from 'ramda/src/path';

import { addNotification } from '../../ducks/notifications';
import { updateRibbon } from '../../ducks/ribbon';
import {
  cancelLegalReferral,
  getLegalReferral,
  setLegalReferralApproved,
} from '../../ducks/legalReferral';
import getHelpers from '../../util/stateHelpers';
import downloadFile from '../../util/downloadFile';
import { isUserSeniorManager } from '../../util/userRoles';
import { status as legalReferralCaseStatus } from '../../constants/legalReferral';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);
  const {
    arrearsDetails: { heading: arrearsDetailsHeading },
    approvelegalCaseReferral: {
      errorText,
      heading,
      labels,
      notification,
      subtitle,
      requiresNonCreatorApproval,
    },
  } = state.dictionary;

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const submissionId = path(['match', 'params', 'submissionId'], ownProps);
  const redirectToDetailsPage = () => ownProps.history.push(`/arrears-details/${arrearsId}`);

  const isLegalReferralCreatedByLoggedUser =
    get(['legalReferral', 'detail', 'owner', 'id']) === get(['user', 'profile', 'id']);
  const isLoggedUserManagerOrSeniorManager = get(['user', 'userType']) === 'manager';

  const isLoggedUserSeniorManager = isUserSeniorManager(get(['user', 'roles']));
  const isLegalReferralWaitingForManagerSignOff =
    get(['legalReferral', 'detail', 'progressStatus']) ===
    legalReferralCaseStatus.pendingSubmission;

  const canApproveReferral =
    !isLegalReferralCreatedByLoggedUser &&
    isLoggedUserManagerOrSeniorManager &&
    !(isLegalReferralWaitingForManagerSignOff && isLoggedUserSeniorManager);
  const showReadOnlyView = !canApproveReferral;
  const showReadOnlyViewNotification = showReadOnlyView && isLegalReferralCreatedByLoggedUser;

  return {
    activePause: get(['pause', 'activePause']),
    arrearsId,
    breadcrumb: getBreadcrumbs(
      heading,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    errorText,
    formError: get(['pause', 'error']),
    heading,
    labels,
    loading: get(['legalReferral', 'loading']),
    notification,
    referralPack: get(['legalReferral', 'detail', 'referralPack']),
    submissionId,
    subtitle,
    redirectToDetailsPage,
    requiresNonCreatorApproval,
    showReadOnlyView,
    showReadOnlyViewNotification,
  };
};

export const mergeProps = (
  {
    breadcrumb,
    heading,
    notification: { approvedTitle, cancelledTitle },
    submissionId,
    redirectToDetailsPage,
    ...stateProps
  },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  getLegalReferral: () => dispatch(getLegalReferral(submissionId)),
  heading,
  onBack: () => redirectToDetailsPage(),
  onSubmit: noteText =>
    dispatch(
      noteText
        ? cancelLegalReferral(submissionId, { noteText })
        : setLegalReferralApproved(submissionId)
    )
      .then(() => {
        redirectToDetailsPage();
        dispatch(
          addNotification({
            dataBddPrefix: 'approveLegalReferral',
            icon: 'success',
            notificationType: 'confirmation',
            title: noteText ? cancelledTitle : approvedTitle,
          })
        );
      })
      .catch(() => {}),
  updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
  downloadFile: (uri, filename) => downloadFile(uri, filename),
  ...ownProps,
});

export default connect(mapStateToProps, null, mergeProps);
