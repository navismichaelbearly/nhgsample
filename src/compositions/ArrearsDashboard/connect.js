import clone from 'ramda/src/clone';
import { connect } from 'react-redux';
import isEmpty from 'ramda/src/isEmpty';

import { getArrearsSummary } from '../../ducks/arrears';
import {
  invalidateSummaryTasks,
  getSummaryTasks,
  getTasksSummary,
  getUserTasksSummary,
} from '../../ducks/tasks';
import getHelpers from '../../util/stateHelpers';
import { updateRibbon } from '../../ducks/ribbon';
import { showPatchSelect } from '../../ducks/patch';
import { sortPatchList, flattenPatches, flattenUsers } from '../../util/patch';

export const mapStateToProps = state => {
  const { get, getBreadcrumbs, getString } = getHelpers(state);

  const profile = get(['user', 'profile']);

  return {
    arrears: get(['arrears', 'items']),
    breadcrumb: getBreadcrumbs(),
    cardLabels: getString(['arrearsSummaryCards']),
    filterLabel: get(['arrears', 'filterLabel']),
    tasksNoDataMsg: getString(['tasksNoDataMsg']),
    myTasksNoDataMsg: getString(['myTasksNoDataMsg']),
    selectedUserTasksNoDataMsg: getString(['selectedUserTasksNoDataMsg']),
    arrearsNoDataMsg: getString(['arrearsNoDataMsg']),
    taskCardLabels: getString(['taskSummaryCards']),
    heading: getString(['arrearsDashboard', 'heading']),
    loadingArrears: get(['arrears', 'loadingArrears']),
    loadingTasks: get(['tasks', 'loading']),
    viewMoreLoading: get(['tasks', 'viewMoreLoading']),
    patches: clone(sortPatchList(get(['patch', 'patchList']), profile)),
    profile,
    resultsCount: get(['arrears', 'resultsCount']),
    selectedUserTasksCount: get(['tasks', 'userItemsCount'], 0),
    youText: getString(['patchSelect', 'you']),
    selectedUserTasks: get(['tasks', 'userItems'], []),
    myTasks: get(['tasks', 'mySummaryItems']),
    tasks: get(['tasks', 'summaryItems']),
  };
};

export const mergeProps = (
  { breadcrumb, heading, patches, profile, ...stateProps },
  { dispatch },
  ownProps
) => {
  const getSummary = (opts, isMine /* used when querying for 'My tasks' */) =>
    dispatch(
      getTasksSummary(
        {
          statuses: 'open',
          EntityType: 'Arrears',
          ...opts,
        },
        isMine
      )
    );
  const flattedUsers = !isEmpty(flattenUsers(patches))
    ? flattenUsers(patches)
        .split(',')
        .filter(e => e !== profile.id)
    : [];
  const userParams = {
    owners: flattedUsers.toString(),
    statuses: 'open',
    EntityType: 'Arrears',
    pageSize: 6,
  };
  const flattenPatchesResults = flattenPatches(patches);
  const patchParams = {
    Patches: flattenPatchesResults,
    statuses: 'open',
    EntityType: 'Arrears',
    pageSize: 6,
  };
  return {
    ...stateProps,
    profile,
    patches,
    getArrearsSummary: opts =>
      dispatch(getArrearsSummary({ patches: flattenPatchesResults, ...opts })),
    getTasksSummary: opts => dispatch(getTasksSummary({ ...opts })),
    getUserTasksSummary: () => {
      dispatch(getUserTasksSummary({ ...userParams }));
    },
    getMyTasksSummary: () => getSummary({ userId: profile.id }, true),
    getSelectedUserTasks: opts => {
      const args = { ...userParams, ...opts };
      dispatch(getUserTasksSummary({ ...args }));
    },
    getSummaryTasks: opts => {
      const args = { ...patchParams, ...opts };
      dispatch(getSummaryTasks({ ...args }));
    },
    onOpenPatchSelect: () => dispatch(showPatchSelect()),
    invalidateTasks: () => dispatch(invalidateSummaryTasks()),
    updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
    ...ownProps,
  };
};

export default connect(mapStateToProps, null, mergeProps);
