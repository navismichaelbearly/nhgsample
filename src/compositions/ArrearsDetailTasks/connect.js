import { connect } from 'react-redux';
import path from 'ramda/src/path';
import pickAll from 'ramda/src/pickAll';

import { setModalContent } from '../../ducks/modal';
import { getTasks, invalidateTasks } from '../../ducks/tasks';
import getHelpers from '../../util/stateHelpers';
import { isUserSeniorManager } from '../../util/userRoles';
import { CANCELLED, COMPLETED, OPEN } from '../../constants/taskStatuses';

export const mapStateToProps = (state, ownProps) => {
  const { get } = getHelpers(state);
  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const isLoading = get(['tasks', 'loading']);
  const {
    arrearsDetails: {
      tasks: { heading, labels },
    },
    archivedTasks,
  } = state.dictionary;
  const user = pickAll(['roles', 'profile'], get(['user']));

  const arrears = get(['arrears', 'items']);
  const currentArrear = arrears.find(arrear => arrear.id === arrearsId) || {};
  const canCreateLegalReferral =
    !isUserSeniorManager(user.roles) &&
    path(['legalReferralCase', 'canCreateLegalReferral'], currentArrear);
  const legalReferralActionStatus = path(['legalReferralCase', 'status'], currentArrear);
  const legalReferralId = path(['legalReferral', 'caseId'], currentArrear);

  return {
    archivedTasksLabels: archivedTasks,
    canCreateLegalReferral,
    legalReferralId,
    legalReferralActionStatus,
    arrearsId,
    isLoading,
    tasks: get(['tasks', 'items']),
    heading,
    labels,
    user,
  };
};

export const mergeProps = ({ arrearsId, ...stateProps }, { dispatch }, ownProps) => ({
  ...stateProps,
  arrearsId,
  invalidateTasks: () => dispatch(invalidateTasks()),
  openTaskModal: modalContent => {
    if (modalContent) {
      dispatch(setModalContent(modalContent));
    }
  },
  getArrearTasks: (archived, opts) =>
    dispatch(
      getTasks({
        EntityId: arrearsId,
        EntityType: 'Arrears',
        Statuses: archived ? `${CANCELLED},${COMPLETED}` : OPEN,
        skipFetchRoles: isUserSeniorManager(stateProps.user.roles),
        ...opts,
      })
    ),
  ...ownProps,
});

export default connect(mapStateToProps, null, mergeProps);
