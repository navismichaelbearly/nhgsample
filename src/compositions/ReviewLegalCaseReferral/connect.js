import { connect } from 'react-redux';
import path from 'ramda/src/path';
import pathOr from 'ramda/src/pathOr';

import getHelpers from '../../util/stateHelpers';
import { updateRibbon } from '../../ducks/ribbon';
import {
  cancelLegalReferral,
  getLegalReferral,
  setLegalReferralSubmitted,
  clear as clearState,
} from '../../ducks/legalReferral';
import { addNotification } from '../../ducks/notifications';
import downloadFile from '../../util/downloadFile';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);
  const {
    arrearsDetails: { heading: arrearsDetailsHeading },
    legalCaseReferral: { formError, legalCaseReferral, reviewReferral },
  } = state.dictionary;

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const submissionId = path(['match', 'params', 'submissionId'], ownProps);
  const redirectToDetailsPage = () => ownProps.history.push(`/arrears-details/${arrearsId}`);

  const arrearsItems = get(['arrears', 'items']);
  const arrearsDetail = arrearsItems.find(i => i.id === arrearsId);
  const requiresApproval = pathOr(true, ['legalReferralCase', 'requiresApproval'], arrearsDetail);
  const notificationSubmittedText = requiresApproval
    ? reviewReferral.notification.submittedText
    : reviewReferral.notification.submittedNonApprovalText;

  return {
    breadcrumb: getBreadcrumbs(
      legalCaseReferral,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    errorStatus: get(['legalReferral', 'errorStatus']),
    formError,
    heading: legalCaseReferral,
    loading: get(['legalReferral', 'loading']),
    notification: reviewReferral.notification,
    notificationSubmittedText,
    redirectToDetailsPage,
    referralPack: get(['legalReferral', 'detail', 'referralPack']),
    requiresApproval,
    submissionId,
    text: { ...reviewReferral },
  };
};

export const mergeProps = (
  {
    breadcrumb,
    heading,
    notification,
    notificationSubmittedText,
    redirectToDetailsPage,
    submissionId,
    ...stateProps
  },
  { dispatch },
  ownProps
) => {
  const dispatchNotification = (dataBddPrefix, title, lines) => {
    dispatch(
      addNotification({
        dataBddPrefix,
        icon: 'success',
        lines,
        notificationType: 'confirmation',
        title,
      })
    );
    redirectToDetailsPage();
  };

  return {
    ...stateProps,
    onSubmit: () =>
      dispatch(setLegalReferralSubmitted(submissionId))
        .then(() => dispatchNotification('submitLegalCaseReferral', notificationSubmittedText))
        .catch(() => {}),
    onCancel: payload =>
      dispatch(cancelLegalReferral(submissionId, payload))
        .then(() =>
          dispatchNotification('cancelLegalCaseReferral', notification.cancelledText, [
            { text: notification.cancelledLine1 },
          ])
        )
        .catch(() => {}),
    updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
    getLegalReferral: () => dispatch(getLegalReferral(submissionId)),
    downloadFile: (uri, filename) => downloadFile(uri, filename),
    invalidateLegalReferralState: () => dispatch(clearState()),
    ...ownProps,
  };
};

export default connect(mapStateToProps, null, mergeProps);
