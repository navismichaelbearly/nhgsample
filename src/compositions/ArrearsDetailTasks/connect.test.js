import * as modalActions from '../../ducks/modal';
import * as tasksActions from '../../ducks/tasks';
import { mapStateToProps, mergeProps } from './connect';
import state from '../../../__mocks__/state';
import { OPEN, CANCELLED, COMPLETED } from '../../constants/taskStatuses';
import * as userRoles from '../../util/userRoles';

jest.mock('../../util/userRoles', () => ({
  isUserSeniorManager: jest.fn(),
}));

describe('ArrearsDashboard connector', () => {
  let mockState;
  let ownProps;

  beforeEach(() => {
    mockState = state();
    modalActions.setModalContent = jest.fn();
    tasksActions.getTasks = jest.fn();
    tasksActions.invalidateTasks = jest.fn();
    ownProps = {
      match: {
        params: {
          arrearsId: 'foo',
        },
      },
      bar: 'baz',
    };
    userRoles.isUserSeniorManager.mockReturnValue(false);
  });

  describe('mapStateToProps', () => {
    it('generates the correct props', () => {
      expect(mapStateToProps(mockState, ownProps)).toMatchSnapshot();
    });
    it('generates the correct props when canCreateLegalReferral is true', () => {
      mockState.arrears.items[0].legalReferralCase.canCreateLegalReferral = true;
      expect(mapStateToProps(mockState, ownProps)).toMatchSnapshot();
    });
    it('generates canCreateLegalReferral=false when user is SeniorManager', () => {
      mockState.arrears.items[0].legalReferralCase.canCreateLegalReferral = true;
      userRoles.isUserSeniorManager.mockReturnValue(true);
      expect(mapStateToProps(mockState, ownProps)).toMatchSnapshot();
    });
  });

  describe('mergeProps', () => {
    let dispatchProps;
    let result;

    beforeEach(() => {
      dispatchProps = {
        dispatch: jest.fn(),
      };
      const mapStateToPropsResult = mapStateToProps(mockState, ownProps);
      result = mergeProps(mapStateToPropsResult, dispatchProps, ownProps);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    it('generates the correct props', () => {
      expect(result).toMatchSnapshot();
    });

    it('performs the correct action when invalidate fires', () => {
      result.invalidateTasks();
      expect(dispatchProps.dispatch).toHaveBeenCalled();
    });

    describe('openTaskModal', () => {
      it('performs the correct actions when openTaskModal fires', () => {
        const modalContent = 'template content';
        result.openTaskModal(modalContent);
        expect(dispatchProps.dispatch).toHaveBeenCalled();
        expect(modalActions.setModalContent).toHaveBeenCalledWith(modalContent);
      });
    });

    it('performs the correct actions when getArrearTasks fires for Open Tasks', () => {
      result.getArrearTasks(false, { rock: 'roll' });
      expect(dispatchProps.dispatch).toHaveBeenCalled();
      expect(tasksActions.getTasks).toHaveBeenCalledWith({
        EntityId: 'foo',
        EntityType: 'Arrears',
        Statuses: OPEN,
        skipFetchRoles: false,
        rock: 'roll',
      });
      expect(userRoles.isUserSeniorManager).toHaveBeenCalledWith([]);
    });

    it('performs the correct actions when getArrearTasks fires for Archived Tasks', () => {
      result.getArrearTasks(true, { rock: 'roll' });
      expect(dispatchProps.dispatch).toHaveBeenCalled();
      expect(tasksActions.getTasks).toHaveBeenCalledWith({
        EntityId: 'foo',
        EntityType: 'Arrears',
        Statuses: `${CANCELLED},${COMPLETED}`,
        skipFetchRoles: false,
        rock: 'roll',
      });
      expect(userRoles.isUserSeniorManager).toHaveBeenCalledWith([]);
    });
  });
});
