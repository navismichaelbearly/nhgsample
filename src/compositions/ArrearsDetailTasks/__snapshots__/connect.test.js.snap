// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrearsDashboard connector mapStateToProps generates canCreateLegalReferral=false when user is SeniorManager 1`] = `
Object {
  "archivedTasksLabels": Object {
    "breadcrumbs": "Closed and cancelled tasks",
    "cancelled": "Cancelled",
    "completed": "Completed",
    "heading": "Closed and cancelled tasks ({count})",
    "reasons": "Reasons for closing",
  },
  "arrearsId": "foo",
  "canCreateLegalReferral": false,
  "heading": "Open Tasks ({count})",
  "isLoading": false,
  "labels": Object {
    "created": "Created",
    "doDate": "Do on",
    "dueDate": "Due by",
    "overdue": "(OVERDUE)",
    "owner": "Owner",
    "reason": "Reason for {reason}",
    "reasons": Object {
      "cancellation": "cancelling",
      "completion": "completing",
      "reassignment": "reassignment",
    },
    "taskCancelButton": "Cancel",
    "taskCompleteButton": "Complete",
    "taskNoButton": "No",
    "taskReassignButton": "Reassign",
    "taskYesButton": "Yes",
    "viewOtherTasks": "View closed and cancelled tasks",
  },
  "legalReferralActionStatus": undefined,
  "legalReferralId": undefined,
  "tasks": Array [
    Object {
      "dueDate": "2018-08-27T15:08:44+00:00",
      "id": "foo",
      "owner": Object {
        "id": "9e0dca1e-0538-e811-80c9-005056825b41",
        "name": "DynamicsAATest Test",
        "type": "systemusers",
      },
      "regardingObjectModel": Object {
        "id": "bar",
        "type": "Arrears",
      },
      "status": "Open",
      "title": "this is a title",
      "type": "manual",
    },
  ],
  "user": Object {
    "profile": Object {
      "createdAt": "2018-01-17T10:31:48.209Z",
      "emailAddress": "mbrownie@test.com",
      "fullname": "Mark Brownie",
      "id": "auth0|5a5f2614de16214961915a31",
      "lastLoginAt": "2018-03-26T11:04:23.262Z",
      "name": "Mark Brownie",
      "patchName": "ABC123",
      "pictureUrl": "https://s.gravatar.com/avatar/b4711b09c3f253d6999f146c91c7d057?s=480&r=pg&d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fmb.png",
      "updatedAt": "2018-03-26T11:04:23.262Z",
      "userIdentities": Object {
        "hasAtLeastOneEmailIdentity": true,
        "hasOnlyEmailIdentities": true,
        "identities": Array [
          Object {
            "connection": "Username-Password-Authentication",
            "isSocial": false,
            "provider": "auth0",
            "userId": "5a5f2614de16214961915a31",
          },
        ],
      },
      "userProfile": Object {
        "assetPartyId": "12345",
        "crmContactId": "71f86a99-03f6-e711-80c5-005056825b41",
        "dateOfBirth": "1995-02-01T00:00:00Z",
        "firstName": "Mark",
        "gender": null,
        "lastName": "Brownie",
      },
    },
    "roles": Array [],
  },
}
`;

exports[`ArrearsDashboard connector mapStateToProps generates the correct props 1`] = `
Object {
  "archivedTasksLabels": Object {
    "breadcrumbs": "Closed and cancelled tasks",
    "cancelled": "Cancelled",
    "completed": "Completed",
    "heading": "Closed and cancelled tasks ({count})",
    "reasons": "Reasons for closing",
  },
  "arrearsId": "foo",
  "canCreateLegalReferral": undefined,
  "heading": "Open Tasks ({count})",
  "isLoading": false,
  "labels": Object {
    "created": "Created",
    "doDate": "Do on",
    "dueDate": "Due by",
    "overdue": "(OVERDUE)",
    "owner": "Owner",
    "reason": "Reason for {reason}",
    "reasons": Object {
      "cancellation": "cancelling",
      "completion": "completing",
      "reassignment": "reassignment",
    },
    "taskCancelButton": "Cancel",
    "taskCompleteButton": "Complete",
    "taskNoButton": "No",
    "taskReassignButton": "Reassign",
    "taskYesButton": "Yes",
    "viewOtherTasks": "View closed and cancelled tasks",
  },
  "legalReferralActionStatus": undefined,
  "legalReferralId": undefined,
  "tasks": Array [
    Object {
      "dueDate": "2018-08-27T15:08:44+00:00",
      "id": "foo",
      "owner": Object {
        "id": "9e0dca1e-0538-e811-80c9-005056825b41",
        "name": "DynamicsAATest Test",
        "type": "systemusers",
      },
      "regardingObjectModel": Object {
        "id": "bar",
        "type": "Arrears",
      },
      "status": "Open",
      "title": "this is a title",
      "type": "manual",
    },
  ],
  "user": Object {
    "profile": Object {
      "createdAt": "2018-01-17T10:31:48.209Z",
      "emailAddress": "mbrownie@test.com",
      "fullname": "Mark Brownie",
      "id": "auth0|5a5f2614de16214961915a31",
      "lastLoginAt": "2018-03-26T11:04:23.262Z",
      "name": "Mark Brownie",
      "patchName": "ABC123",
      "pictureUrl": "https://s.gravatar.com/avatar/b4711b09c3f253d6999f146c91c7d057?s=480&r=pg&d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fmb.png",
      "updatedAt": "2018-03-26T11:04:23.262Z",
      "userIdentities": Object {
        "hasAtLeastOneEmailIdentity": true,
        "hasOnlyEmailIdentities": true,
        "identities": Array [
          Object {
            "connection": "Username-Password-Authentication",
            "isSocial": false,
            "provider": "auth0",
            "userId": "5a5f2614de16214961915a31",
          },
        ],
      },
      "userProfile": Object {
        "assetPartyId": "12345",
        "crmContactId": "71f86a99-03f6-e711-80c5-005056825b41",
        "dateOfBirth": "1995-02-01T00:00:00Z",
        "firstName": "Mark",
        "gender": null,
        "lastName": "Brownie",
      },
    },
    "roles": Array [],
  },
}
`;

exports[`ArrearsDashboard connector mapStateToProps generates the correct props when canCreateLegalReferral is true 1`] = `
Object {
  "archivedTasksLabels": Object {
    "breadcrumbs": "Closed and cancelled tasks",
    "cancelled": "Cancelled",
    "completed": "Completed",
    "heading": "Closed and cancelled tasks ({count})",
    "reasons": "Reasons for closing",
  },
  "arrearsId": "foo",
  "canCreateLegalReferral": true,
  "heading": "Open Tasks ({count})",
  "isLoading": false,
  "labels": Object {
    "created": "Created",
    "doDate": "Do on",
    "dueDate": "Due by",
    "overdue": "(OVERDUE)",
    "owner": "Owner",
    "reason": "Reason for {reason}",
    "reasons": Object {
      "cancellation": "cancelling",
      "completion": "completing",
      "reassignment": "reassignment",
    },
    "taskCancelButton": "Cancel",
    "taskCompleteButton": "Complete",
    "taskNoButton": "No",
    "taskReassignButton": "Reassign",
    "taskYesButton": "Yes",
    "viewOtherTasks": "View closed and cancelled tasks",
  },
  "legalReferralActionStatus": undefined,
  "legalReferralId": undefined,
  "tasks": Array [
    Object {
      "dueDate": "2018-08-27T15:08:44+00:00",
      "id": "foo",
      "owner": Object {
        "id": "9e0dca1e-0538-e811-80c9-005056825b41",
        "name": "DynamicsAATest Test",
        "type": "systemusers",
      },
      "regardingObjectModel": Object {
        "id": "bar",
        "type": "Arrears",
      },
      "status": "Open",
      "title": "this is a title",
      "type": "manual",
    },
  ],
  "user": Object {
    "profile": Object {
      "createdAt": "2018-01-17T10:31:48.209Z",
      "emailAddress": "mbrownie@test.com",
      "fullname": "Mark Brownie",
      "id": "auth0|5a5f2614de16214961915a31",
      "lastLoginAt": "2018-03-26T11:04:23.262Z",
      "name": "Mark Brownie",
      "patchName": "ABC123",
      "pictureUrl": "https://s.gravatar.com/avatar/b4711b09c3f253d6999f146c91c7d057?s=480&r=pg&d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fmb.png",
      "updatedAt": "2018-03-26T11:04:23.262Z",
      "userIdentities": Object {
        "hasAtLeastOneEmailIdentity": true,
        "hasOnlyEmailIdentities": true,
        "identities": Array [
          Object {
            "connection": "Username-Password-Authentication",
            "isSocial": false,
            "provider": "auth0",
            "userId": "5a5f2614de16214961915a31",
          },
        ],
      },
      "userProfile": Object {
        "assetPartyId": "12345",
        "crmContactId": "71f86a99-03f6-e711-80c5-005056825b41",
        "dateOfBirth": "1995-02-01T00:00:00Z",
        "firstName": "Mark",
        "gender": null,
        "lastName": "Brownie",
      },
    },
    "roles": Array [],
  },
}
`;

exports[`ArrearsDashboard connector mergeProps generates the correct props 1`] = `
Object {
  "archivedTasksLabels": Object {
    "breadcrumbs": "Closed and cancelled tasks",
    "cancelled": "Cancelled",
    "completed": "Completed",
    "heading": "Closed and cancelled tasks ({count})",
    "reasons": "Reasons for closing",
  },
  "arrearsId": "foo",
  "bar": "baz",
  "canCreateLegalReferral": undefined,
  "getArrearTasks": [Function],
  "heading": "Open Tasks ({count})",
  "invalidateTasks": [Function],
  "isLoading": false,
  "labels": Object {
    "created": "Created",
    "doDate": "Do on",
    "dueDate": "Due by",
    "overdue": "(OVERDUE)",
    "owner": "Owner",
    "reason": "Reason for {reason}",
    "reasons": Object {
      "cancellation": "cancelling",
      "completion": "completing",
      "reassignment": "reassignment",
    },
    "taskCancelButton": "Cancel",
    "taskCompleteButton": "Complete",
    "taskNoButton": "No",
    "taskReassignButton": "Reassign",
    "taskYesButton": "Yes",
    "viewOtherTasks": "View closed and cancelled tasks",
  },
  "legalReferralActionStatus": undefined,
  "legalReferralId": undefined,
  "match": Object {
    "params": Object {
      "arrearsId": "foo",
    },
  },
  "openTaskModal": [Function],
  "tasks": Array [
    Object {
      "dueDate": "2018-08-27T15:08:44+00:00",
      "id": "foo",
      "owner": Object {
        "id": "9e0dca1e-0538-e811-80c9-005056825b41",
        "name": "DynamicsAATest Test",
        "type": "systemusers",
      },
      "regardingObjectModel": Object {
        "id": "bar",
        "type": "Arrears",
      },
      "status": "Open",
      "title": "this is a title",
      "type": "manual",
    },
  ],
  "user": Object {
    "profile": Object {
      "createdAt": "2018-01-17T10:31:48.209Z",
      "emailAddress": "mbrownie@test.com",
      "fullname": "Mark Brownie",
      "id": "auth0|5a5f2614de16214961915a31",
      "lastLoginAt": "2018-03-26T11:04:23.262Z",
      "name": "Mark Brownie",
      "patchName": "ABC123",
      "pictureUrl": "https://s.gravatar.com/avatar/b4711b09c3f253d6999f146c91c7d057?s=480&r=pg&d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fmb.png",
      "updatedAt": "2018-03-26T11:04:23.262Z",
      "userIdentities": Object {
        "hasAtLeastOneEmailIdentity": true,
        "hasOnlyEmailIdentities": true,
        "identities": Array [
          Object {
            "connection": "Username-Password-Authentication",
            "isSocial": false,
            "provider": "auth0",
            "userId": "5a5f2614de16214961915a31",
          },
        ],
      },
      "userProfile": Object {
        "assetPartyId": "12345",
        "crmContactId": "71f86a99-03f6-e711-80c5-005056825b41",
        "dateOfBirth": "1995-02-01T00:00:00Z",
        "firstName": "Mark",
        "gender": null,
        "lastName": "Brownie",
      },
    },
    "roles": Array [],
  },
}
`;
